rust/src/lib.rs

```diff
@@ -547,16 +576,15 @@ pub unsafe extern "C" fn zcashlc_init_blocks_table(
         let sapling_tree =
             hex::decode(unsafe { CStr::from_ptr(sapling_tree_hex) }.to_str()?).unwrap();
 
-        match init_blocks_table(
+        init_blocks_table(
             &db_data,
             BlockHeight::from_u32(height as u32),
             hash,
             time,
             &sapling_tree,
-        ) {
-            Ok(()) => Ok(1),
-            Err(e) => Err(format_err!("Error while initializing blocks table: {}", e)),
-        }
+        )
+        .map(|()| 1)
+        .with_context(|| "Error while initializing blocks table")
```

Comment:
This actually breaks the error we expect, because the database throws an "everything is fine" error here